Embedded Dashboard Panels: Add Grafana Branding

nmarrs opened the pull request:
**What is this feature?**

Adds a “Powered by Grafana” branding overlay to `d-solo` (solo panel embed) pages via `SoloPanelPageLogo`. The overlay scales with panel size, hides on hover, and can be hidden via a `hideLogo` query param. Panel image generation URLs now include `hideLogo=true` so rendered images don’t include the overlay.

Non responsive design (from early dev of feature)
<img width="500" height="234" alt="before resize" src="https://github.com/user-attachments/assets/07d4cb91-1a80-46ef-8488-0de58ef21bfb" />


Responsive design (current)
<img width="500" height="238" alt="auto-resize (now smaller)" src="https://github.com/user-attachments/assets/80de9e17-f591-4308-b1f0-984cdb42db3d" />

<img width="501" height="289" alt="auto-resize (small) after" src="https://github.com/user-attachments/assets/a360b23e-ad46-4c22-91be-32ddbca28cf6" />


**Why do we need this feature?**

To improve Grafana brand visibility for embedded panels while still avoiding interference with panel content and ensuring image rendering/export workflows can generate clean (unbranded) images when required.

**Who is this feature for?**

- Users embedding single panels in iframes.


**Feature in action**


https://github.com/user-attachments/assets/9168e381-9a58-4a90-94d3-f2f03ac09f8e

nmarrs committed:
feat: add Grafana logo to embedded panels

- Add Grafana logo watermark to solo panel view (embedded panels)
- Logo appears in top-right corner with subtle background container
- Logo hides on hover to avoid interfering with panel content
- Uses React state to track hover for reliable behavior across nested elements

nmarrs committed:
minor formatting

nmarrs committed:
Merge remote-tracking branch 'origin' into feature/embedded-panel-logo

nmarrs committed:
update changes to match public dashboards styling

nmarrs committed:
match styles of public dashboards

nmarrs committed:
erge remote-tracking branch 'origin' into feature/embedded-panel-logo

nmarrs committed:
feat: add responsive Grafana branding to embedded panels

- Add 'Powered by Grafana' branding with text logo to solo panel view
- Implement responsive scaling based on panel dimensions (0.6x to 1.0x)
- Logo and text scale proportionally with panel size
- Branding hides on hover to avoid interfering with panel content
- Matches public dashboard branding pattern for consistency
- Uses ResizeObserver for efficient responsive updates

nmarrs committed:
feat: add Grafana branding to embedded solo panels

- Add 'Powered by Grafana' branding with text logo to embedded panels
- Create SoloPanelPageLogo component for reusable branding
- Implement responsive scaling based on panel dimensions
- Add hover-to-hide functionality to avoid content overlap
- Logo scales between 0.6x and 1.0x based on panel size

nmarrs committed:
refactor: move scale calculation into SoloPanelPageLogo component

- Move responsive scale calculation logic from SoloPanelRenderer to SoloPanelPageLogo
- Logo component now manages its own scaling based on container dimensions
- Improves separation of concerns and component encapsulation

nmarrs committed:
feat: add hideLogo query parameter to disable embedded panel branding

- Add hideLogo query parameter support to SoloPanelPage
- Logo can be hidden via ?hideLogo, ?hideLogo=true, or ?hideLogo=1
- Useful for customers who want to disable branding and for image rendering scenarios
- Update Props interface to include hideLogo in queryParams type

nmarrs committed:
feat: hide logo in panel image renderer URLs

- Add hideLogo=true parameter to image renderer URLs in ShareLinkTab
- Ensures logo is hidden when generating panel images through share feature
- Update test to expect hideLogo=true in render URL

nmarrs committed:
feat: hide logo in old dashboard sharing panel image URLs

- Add hideLogo=true parameter to buildImageUrl in ShareModal utils
- Ensures logo is hidden when generating panel images through old share modal
- Update all ShareLink tests to expect hideLogo=true in render URLs

nmarrs committed:
test: add comprehensive tests for SoloPanelPage and SoloPanelPageLogo

- Add SoloPanelPageLogo tests covering rendering, hover behavior, theme selection, and scaling
- Add SoloPanelPage tests covering logo visibility based on hideLogo prop
- Test logo hiding functionality (most important behavior)
- Test responsive scaling based on container dimensions
- Test ResizeObserver integration
- All 14 tests passing

nmarrs committed:
refactor: centralize hideLogo handling in SoloPanelPageLogo

Move hideLogo parsing and decision-making into SoloPanelPageLogo so SoloPanelPage/SoloPanelRenderer only pass through the raw query param value.

nmarrs committed:
chore: clean up solo logo test and share link params

Remove a duplicate SVG mock in SoloPanelPageLogo.test, and simplify ShareLinkTab image URL building without changing behavior.

nmarrs committed:
chore: revert ShareLinkTab image query refactor

Restore the previous image URL query-param mutation logic in ShareLinkTab to reduce risk.

nmarrs committed:
chore: set hideLogo once for ShareLinkTab image URLs

Avoid passing hideLogo twice when building the rendered image URL.

nmarrs committed:
fix: handle boolean hideLogo query param in SoloPanelPageLogo

Handle query params that are represented as booleans (e.g., ?hideLogo) and arrays, and avoid calling trim() on non-strings.

nmarrs committed:
fix i18n

evictorero reviewed (approved):
LGTM. I've tested it with different sizes and themes, and it disappears when hovering over it.
Asked Copilot to review it. Mostly to take a look at the math and css classes and styles.
I'm wondering if we should add this new queryParam to the docs to document how to hide it if needed.

nmarrs committed:
Merge remote-tracking branch 'origin' into feature/embedded-panel-logo

nmarrs committed:
fix(dashboard-scene): address SoloPanelPageLogo review feedback

Avoid double-scaling logo margin, clarify scaling comments, and extend tests for null/array values and ResizeObserver cleanup.

nmarrs committed:
update margin left on logo to better match text spacing
